R/create_model_design.R

```diff
@@ -38,21 +37,23 @@ create_model_design <- function(
   ruv = NULL
 ) {
   design <- list()
+  params <- c("model", "parameters", "omega_matrix", "ruv")
   if(!is.null(lib)) {
     suppressMessages( ## avoid message "the following objects are masked from ..."
       require(lib, character.only = TRUE)
     )
-    load_lib <- c("model", "parameters", "omega_matrix", "ruv")
-    for(i in seq(load_lib)) {
-      design[[load_lib[i]]] <- get(load_lib[i], asNamespace(lib))()
+    for(i in seq(params)) {
+      design[[params[i]]] <- get(params[i], asNamespace(lib))()
     }
   } else {
-    design <- list(
-      model = model,
-      parameters = parameters,
-      omega_matrix = omega_matrix,
-      ruv = ruv
-    )
+    design <- list()
+  }
+  ## potentially override using user-specified details
+  for(key in params) {
```

Comment:
this change makes it possible to use a standard model, but override only e.g. the `ruv` (without having to respecify all other details)